e2e/cases/server/public-dir/publicDir.test.ts

```diff
@@ -170,6 +170,85 @@ test('should serve publicDir for preview server correctly', async ({
   await rsbuild.close();
 });
 
+test('should copy publicDir to the environment distDir when multiple environments', async () => {
```

Comment:
Can we add some cases for dev too? We need to make sure that dev and build have the same result.

Reply:
We didn't copy public dir in dev, only serve public files to server.base(`/`). Therefore dev will not be affected by multiple environments.